src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Metadata.cs

```diff
@@ -80,9 +80,24 @@ private static Metadata GetMetadataNoThrow(PortableExecutableReference reference
             var metadata = GetMetadataNoThrow(reference);
             if (metadata == null)
             {
-                return null;
+                return SpecializedTasks.Default<SymbolTreeInfo>();
+            }
+
+            // Try to acquire the data outside the lock.  That way 
+            SymbolTreeInfo info;
+            if (s_metadataIdToInfo.TryGetValue(metadata.Id, out info))
```

Comment:
Is this the only place s_metadataIdToInfo is used?  If so, consider storing the Task already in the dictionary, as this still allocates on every `Task.FromResult` call.